config/condition.go

```diff
@@ -0,0 +1,123 @@
+package config
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/consul/lib/decode"
+	"github.com/mitchellh/mapstructure"
+)
+
+// ConditionConfig configures a condition on a task to define the condition on
+// which to execute a task.
+type ConditionConfig interface {
+	Copy() ConditionConfig
+	Merge(ConditionConfig) ConditionConfig
+	Finalize([]string)
+	Validate() error
+	GoString() string
+}
+
+// DefaultConditionConfig returns the default conditions which is an unconfigured
+// 'services' type condition.
+func DefaultConditionConfig() ConditionConfig {
+	return &ServicesConditionConfig{}
+}
+
+// isNil can be used to check if a ConditionConfig interface is nil by checking
+// both the type and value. Not needed for checking a ConditionConfig
+// implementation i.e. isNil(ConditionConfig), servicesConditionConfig == nil
+func isNil(c ConditionConfig) bool {
```

Comment:
Since this is in the `config` package with many types, can you change the function name to be more descriptive to the type it's checking for?

Reply:
Updated 👍